### PR TITLE
Phone metadata v2

### PR DESCRIPTION
**Pull request by ramont-stripe**
## Summary

Implemented the proposal for deriving phone formatting metadata from libphonenumber.

## Motivation

https://paper.dropbox.com/doc/Br2RieMOwzxUaMTgHnMP1fMpAg-76u8OlLRyoGRAIHtabmNU

## Testing

* The previous formatting logic was covered by unit and snapshot tests
* Added new tests
* Manual testing

## Changelog

NA.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# See: https://EditorConfig.org
```
Comment (eurias-stripe) [inline .editorconfig:1]: This file doesn't seem to belong as part of this PR.
We should have common settings, but we shouldn't mix things that are out of scope from a PR

Comment (ramont-stripe) [reply .editorconfig]: 👍 I will open a separate PR. This PR required writing a lot of ruby code and not having default settings became tedious.

File: StripeUICore/StripeUICore/Source/PhoneNumber/PhoneNumber.swift [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+        // Matching regex: ^\+[1-9]\d{2,14}$
+        guard
+            characters.count > 4,
+            characters.count <= Constants.e164MaxDigits + 1,
+            characters[0] == "+",
+            characters[1] != "0",
+            characters[1...].allSatisfy({
+                $0.unicodeScalars.allSatisfy(CharacterSet.stp_asciiDigit.contains(_:))
+            })
```
[the remark this answers is not in the document]
Comment (ramont-stripe) [reply StripeUICore/StripeUICore/Source/PhoneNumber/PhoneNumber.swift:97]: I over-optimized when I originally implemented the E.164 parser. This file is mostly existing code that was moved from one folder into another:

Screen Shot 2022-11-01 at 4 39 22 PM

I plan to revisit E.164 parsing as a follow up.

[the remark this answers is not in the document]
Comment (eurias-stripe) [reply StripeUICore/StripeUICore/Source/PhoneNumber/PhoneNumber.swift:97]: Fair enough.

**Review by eurias-stripe — DISMISSED** [review]

**Review by eurias-stripe — APPROVED** [review]
